subprojects/packagefiles/pugixml/meson.build

```diff
@@ -0,0 +1,61 @@
+project('pugixml', 'cpp', version: '1.11', meson_version: '>=0.46.0')
+
+cxx = meson.get_compiler('cpp')
+
+sources = files('src/pugixml.cpp')
+
+if host_machine.system() == 'windows'
+  win_mod = import('windows')
+  sources += win_mod.compile_resources('scripts/pugixml_dll.rc')
+endif
+
+args = []
+if cxx.get_id() == 'msvc'
+  if get_option('default_library') == 'shared'
+    args +='-DPUGIXML_API=__declspec(dllexport)'
+  elif get_option('default_library') == 'both'
+    error('cannot build both shared and static on MSVC due to declspec')
+  endif
+endif
+
+foreach opt: ['wchar_mode', 'compact', 'no_xpath', 'no_stl', 'no_exceptions']
+  if get_option(opt)
+    args += '-DPUGIXML_' + opt.to_upper()
+  endif
+endforeach
+
+pugixml_lib = library('pugixml', sources,
+  cpp_args: args,
+  version: meson.project_version(),
+  install: true,
+)
+install_headers('src/pugiconfig.hpp', 'src/pugixml.hpp')
+
+pugixml_dep = declare_dependency(
+  link_with: pugixml_lib,
+  include_directories: include_directories('src')
+)
+
+import('pkgconfig').generate(
+  pugixml_lib,
+  description: 'Light-weight, simple and fast XML parser for C++ with XPath support.',
+  url: 'https://pugixml.org/',
+)
+
+if find_program('cmake', required: false).found()
+  cmake = import('cmake')
+  cmake.write_basic_package_version_file(
+    name: meson.project_name(),
+    version: meson.project_version(),
+    compatibility: 'SameMajorVersion',
+  )
+  cmake.configure_package_config_file(
+    input: 'scripts/pugixml-config.cmake.in',
+    name: meson.project_name(),
+    configuration: configuration_data(),
+  )
+endif
+
+subdir('tests')
+pugixml_check = executable('pugixml-check', test_sources, dependencies: pugixml_dep)
+test('pugixml-check', pugixml_check, workdir: meson.current_source_dir())
```

Comment:
Any reason for not moving that into test/ subdir?

Reply:
It actually wants that subdir, and it is easier than playing around with meson versions and global_source_root and suchlike.

Reply:
It's more complicated than `workdir: meson.project_source_root()` ?

Reply:
That would raise the minimum version of meson! Which I don't like to do in general without good reason, and especially since I actually did offer this upstream now.

And I really have no issue with programs being built in the root build dir, so it never crossed my mind to try to make it work like that anyway.

Reply:
ok fair enough, was a small nitpicking anyway.